src/assemble_workflow/public_urls.py

```diff
@@ -0,0 +1,22 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# The OpenSearch Contributors require contributions made to
+# this file be licensed under the Apache-2.0 license or a
+# compatible open source license.
+
+import os
+from urllib.parse import urljoin
+
+class PublicURLpath:
+    def __init__(self, build): 
+        self.public_url = os.getenv("PUBLIC_ARTIFACT_URL", None)
```

Comment:
We should stop using the environment variable completely and move to having a base url passed in via the assembly.sh command which is then passed into the bundle recorder.

By using the environment variable this is effectively a hidden interface to this component, that makes it hard to test and debug.  If we are explicitly passing this value the behavior is much more transparent

Reply:
The thing is do we still support the case that no base url is assigned? Currently if the environment variable `PUBLIC_ARTIFACT_URL` doesn't exist, we will generate the location path based on local directory instead of an URL.

Reply:
This URL is only used to decide what to write in the .yml output as a root so that various artifacts can be directly downloaded later. For local scenarios it's blank, and it's the local directory where we drop artifacts that needs to change to include product name, architecture, and platform.